From 14296d0a19df11b9c89220806f5ace3baf7b5889 Mon Sep 17 00:00:00 2001
In-Reply-To: <20190626000329.32475-1-avarab@gmail.com>
References: <20190626000329.32475-1-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Fri, 28 Jun 2019 01:25:05 +0200
Subject: [PATCH v2 10/10] grep: refactor ifdef macro to use "HAVE_LIBPCRE2"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Introduce and use a HAVE_LIBPCRE2 define instead of having
conditionally compiled code. This follows the example of HAVE_THREADS
added in 0ec79358d0 ("thread-utils: macros to unconditionally compile
pthreads API", 2018-10-27).

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 grep.c | 21 ++++++++++-----------
 grep.h |  2 ++
 2 files changed, 12 insertions(+), 11 deletions(-)

diff --git a/grep.c b/grep.c
index fc0ed73ef3..64e27e00c1 100644
--- a/grep.c
+++ b/grep.c
@@ -214,17 +214,16 @@ static void grep_set_pattern_type_option(enum grep_pattern_type pattern_type, st
 		break;
 
 	case GREP_PATTERN_TYPE_PCRE:
-#ifdef USE_LIBPCRE2
-		opt->pcre2 = 1;
-#else
-		/*
-		 * It's important that pcre1 always be assigned to
-		 * even when there's no USE_LIBPCRE* defined. We still
-		 * call the PCRE stub function, it just dies with
-		 * "cannot use Perl-compatible regexes[...]".
-		 */
-		opt->pcre1 = 1;
-#endif
+		if (HAVE_LIBPCRE2)
+			opt->pcre2 = 1;
+		else
+			/*
+			 * It's important that pcre1 always be assigned to
+			 * even when there's no USE_LIBPCRE* defined. We still
+			 * call the PCRE stub function, it just dies with
+			 * "cannot use Perl-compatible regexes[...]".
+			 */
+			opt->pcre1 = 1;
 		break;
 	}
 }
diff --git a/grep.h b/grep.h
index d35a137fcb..f008c3ea65 100644
--- a/grep.h
+++ b/grep.h
@@ -23,9 +23,11 @@ typedef int pcre_extra;
 typedef int pcre_jit_stack;
 #endif
 #ifdef USE_LIBPCRE2
+#define HAVE_LIBPCRE2 1
 #define PCRE2_CODE_UNIT_WIDTH 8
 #include <pcre2.h>
 #else
+#define HAVE_LIBPCRE2 0
 typedef int pcre2_code;
 typedef int pcre2_match_data;
 typedef int pcre2_compile_context;
-- 
2.22.0.455.g172b71a6c5

